tests/unit/test_supermetrics.py

```diff
@@ -0,0 +1,172 @@
+                    ]
+                }
+            }
+        }
+    # Set the mock to return the mock response object
+    mock_handle_api_response.return_value.json.return_value = mock_response
+
+    # Call the method under test
+    response = supermetrics.to_json()
+
+    # Assert that the response is as expected
+    assert response == {
+        "data": [["value1", "value2"]],
+        "meta": {
+            "query": {
+                "fields": [
+                    {"field_name": "col1"},
+                    {"field_name": "col2"}
+                ]
+            }
+        }
+    }
+
+def test_to_df_with_data(supermetrics, mocker):
+    # Mock the API response with some data
+    mock_response = {
+                "meta": {
+                    "query": {
+                        "ds_id": "GA",  #Data source ID, e.g., Google Analytics
+                        "fields": [
+                            {
+                                "field_name": "date",
+                                "field_type": "DIMENSION",
+                                "field_split": "row"
+                            },
+                            {
+                                "field_name": "sessions",
+                                "field_type": "METRIC",
+                                "field_split": "row"
+                            }
+                        ],
+                        "other_query_metadata": "..."
+                    },
+                    "status": "success",  # Status of the query
+                    "execution_time": "0.456"  # Time taken to execute the query
+                },
+                "data": [
+                    ["2023-01-01", 100],  # Example data rows
+                    ["2023-01-02", 150],
+                    ["2023-01-03", 120]
+                ],
+                "paging": {
+                    "current_page": 1,  # Current page number if pagination is used
+                    "total_pages": 1,
+                    "total_results": 3
+                }
+            }
+
+    mock_method = mocker.patch('viadot.sources.supermetrics.Supermetrics.to_json')
+    mock_method.return_value = mock_response
+    mock_method = mocker.patch('viadot.sources.supermetrics.Supermetrics._get_col_names')
+    mock_method.return_value=["date", "sessions"]
+
+    query_params = {"ds_id": "GA", "query": "test_query"}
+    df = supermetrics.to_df()
+
+    assert not df.empty
+    assert list(df.columns) == ["date", "sessions","_viadot_source", "_viadot_downloaded_at_utc"]
+    
+def test_get_col_names_google_analytics_pivoted(mocker, supermetrics):
+    mock_response = {
+        "meta": {
+            "query": {
+                "fields": [
+                    {"field_name": "ga:date", "field_split": "column"},
+                    {"field_name": "ga:sessions", "field_split": "row"}
+                ]
+            }
+        },
+        "data": [{"ga:date": "2023-01-01", "ga:sessions": 100}]
+    }
```

Comment:
```suggestion
    mock_response = {
        "meta": {
            "query": {
                "fields": [
                    {"field_name": "ga:date", "field_split": "column"},
                    {"field_name": "ga:sessions", "field_split": "row"}
                ]
            }
        },
        "data": [{"ga:date1": "2023-01-01", "ga:date2": "2024-01-01", "ga:sessions": 100}]
    }
```
I think the idea is that in "pivoted" dataset, fields are different between what's in `meta->query->fields` and what's in `data`, and so we take the field names from `data` key as this one is correct. So you need test data like this to actually test this (we expect to get 3 columns as the result).

Reply:
https://github.com/dyvenia/viadot/issues/1059